slicer-plugin/NvidiaAIAA/SegmentEditorNvidiaAIAALib/SegmentEditorEffect.py

```diff
@@ -512,30 +531,43 @@ def onClickDeepgrow(self, current_point):
         if in_file is None and session_id is None:
             return
 
+        # use model name to determine
+        deepgrow_3d = True if '3d' in model.lower() else False
         start = time.time()
 
         label = self.currentSegment().GetName()
-        operationDescription = 'Run Deepgrow for segment: {}; model: {}'.format(label, model)
+        operationDescription = 'Run Deepgrow for segment: {}; model: {}; 3d {}'.format(label, model, deepgrow_3d)
         logging.debug(operationDescription)
+        result = 'FAILED'
 
         try:
             qt.QApplication.setOverrideCursor(qt.Qt.WaitCursor)
 
             sliceIndex = current_point[2]
             logging.debug('Slice Index: {}'.format(sliceIndex))
 
-            foreground = [x for x in foreground_all if x[2] == sliceIndex]
-            background = [x for x in background_all if x[2] == sliceIndex]
+            if deepgrow_3d:
+                foreground = foreground_all
+                background = background_all
+            else:
+                foreground = [x for x in foreground_all if x[2] == sliceIndex]
+                background = [x for x in background_all if x[2] == sliceIndex]
 
             logging.debug('Foreground: {}'.format(foreground))
             logging.debug('Background: {}'.format(background))
+            logging.debug('Current point: {}'.format(current_point))
 
-            result_file = self.logic.deepgrow(in_file, session_id, model, foreground, background)
-            result = 'FAILED'
+            result_file, params = self.logic.deepgrow(in_file, session_id, model, foreground, background, [current_point])
+            logging.debug('Params from deepgrow is {}'.format(params))
```

Comment:
result = 'FAILED' not required? can u check where this value is checked? because below if/else-if has no else condition... so somewhere 'FAILED' value might be used to pop up an alert

Reply:
I did not remove this.
I move this result = 'FAILED' up in line 541 outside the try block.
Cuz we need this outside try block at line 587